components/TimeRange.js

```diff
@@ -126,7 +123,7 @@ function TimeRange({ onChange = () => null, range }) {
         }
       }
     }
-  }, [range]);
+  }, [range, customValue]);
```

Comment:
@yanglin5689446 This is causing lint error because of the use of `setAndUpdateValue`, which is updated every `render`. The logic in this hook is a bit complicated and its side effect (invoking `setAndUpdateValue`) is costly.

Would you take a look at this and determine an appropriate dependency list here?

Reply:
I think it's safe to include only `range` as dependency, because this effect should only run when props passed from parent update, only when this happen the effect should be run and bind to new closure, though I'm not pretty sure if it works as I expected 😅

Reply:
with `customValue` included in deps list, the state in component will always override the props passed from parent so the props update from parent never applied(correct me if I'm wrong)

Reply:
Got it. In the latest change, I did not put `customValue` in the dep list.

However, the use of `setAndUpdateValue` still leaves opportunity that this `useEffect` function can invoke old `onChange` prop from some very old closure. According to react doc, it is [not safe to omit functions from the list of dependencies](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies).

I have re-written all logic in `useEffect` using functional updates of states, as suggested in [React documentation](https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often).  Hopefully it works all the same.